api/dataprocessor_test.go

```diff
@@ -1095,6 +1096,194 @@ func TestMergeSeries(t *testing.T) {
 	}
 }
 
+func TestRequestContextWithoutConsolidator(t *testing.T) {
+	metric := "metric1"
+	archInterval := uint32(10)
+	req := reqRaw(metric, 44, 88, 100, 10, consolidation.None)
+	req.ArchInterval = archInterval
+	ctx := newRequestContext(&req, consolidation.None)
+
+	expectFrom := uint32(41)
+	if ctx.From != expectFrom {
+		t.Errorf("requestContext.From is not at boundary as expected, expected/got %d/%d", expectFrom, ctx.From)
+	}
+
+	expectTo := uint32(81)
+	if ctx.To != expectTo {
+		t.Errorf("requestContext.To is not at boundary as expected, expected/got %d/%d", expectTo, ctx.To)
+	}
+
+	expectedAggKey := ""
+	if ctx.AggKey != expectedAggKey {
+		t.Errorf("requestContext.AggKey is not empty as expected, expected/got \"%s\"/\"%s\"", expectedAggKey, ctx.AggKey)
+	}
+}
+
+func TestRequestContextWithConsolidator(t *testing.T) {
+	metric := "metric1"
+	archInterval := uint32(10)
+	from := uint32(44)
+	to := uint32(88)
+	req := reqRaw(metric, from, to, 100, 10, consolidation.Sum)
+	req.ArchInterval = archInterval
+	ctx := newRequestContext(&req, consolidation.Sum)
+
+	expectFrom := from
+	if ctx.From != expectFrom {
+		t.Errorf("requestContext.From is not original value as expected, expected/got %d/%d", expectFrom, ctx.From)
+	}
+
+	expectTo := to
+	if ctx.To != expectTo {
+		t.Errorf("requestContext.To is not original value as expected, expected/got %d/%d", expectTo, ctx.To)
+	}
+
+	expectedAggKey := fmt.Sprintf("%s_%s_%d", metric, "sum", archInterval)
+	if ctx.AggKey != expectedAggKey {
+		t.Errorf("requestContext.AggKey is not as expected, expected/got \"%s\"/\"%s\"", expectedAggKey, ctx.AggKey)
+	}
+}
+
+func TestGetSeriesCachedStore(t *testing.T) {
```

Comment:
it's hard to understand the bigger picture of this test, can you explain the general idea?
something like:
* create 4 chunks
* add 3 of them to the cache
* add the last of them to the store (why not the other 3?)
...